micro/service.go

```diff
@@ -325,7 +325,7 @@ func (s Verb) String() string {
 // A service name, version and Endpoint configuration are required to add a service.
 // AddService returns a [Service] interface, allowing service management.
 // Each service is assigned a unique ID.
-func AddService(nc *nats.Conn, config Config) (Service, error) {
+func AddService(nc *nats.Conn, config Config) (_ Service, err error) {
```

Comment:
I'm not sure I like having named returns here - they tend to make it harder to debug/understand code. After your other changes, you have it just for the sake of 2 returns (lines 365 and 394). IMO it would be cleaner to simply remove the defer and named returns and just close `asyncDispatcher` in those 2 lines.

Reply:
I personally find using named return parameters/`defer` combo to be an effective and intention-revealing tool (as long as the actual `return` statements are explicit). But in this case fewer lines of code is better? :) Done.